google/cloud/bigquery/dataset.py

```diff
@@ -77,28 +77,29 @@ def _get_routine_reference(self, routine_id):
 class AccessEntry(object):
```

Comment:
I notice this class uses an older pattern of hidden properties such as `_role` and `_entity_type`. I'd much prefer if it followed our `_properties` pattern that most of our other classes use. That way it'll be flexible to new properties in the future (such as during a private preview before we have a chance to implement the feature).

Perhaps best left as a follow-up PR so as not to block the customers who need it shorter term. https://github.com/googleapis/python-bigquery/issues/1077 Let me know if you're interested in that somewhat larger refactor.